### PR TITLE
Add LZ78 Compression and Decompression Implementation with Unit Tests

### PR DESCRIPTION
**Pull request by vaibhavsiingh**
This pull request introduces an implementation of the LZ78 compression algorithm along with its corresponding decompression functionality. Additionally, it includes unit tests.

### Implementation of the LZ78 algorithm:
* [`slp-eval-py/src/slp_eval/lz78.py`](diffhunk://#diff-0d064ff93c8ec1bf9b47156cd6325e992db07f09e219d056eeb1db1e704f3788R1-R48): Added a new `LZ78` class with methods `compress` and `decompress` to perform LZ78 compression and decompression. The implementation uses dictionaries to manage mappings and supports generic types for input data.

### Unit tests for LZ78:
* [`slp-eval-py/src/slp_eval/tests/lz78_test.py`](diffhunk://#diff-5a15b26788458a420b8f3baedf75a40c7c245b0ddcfe86e797938305bce15db5R1-R37): Added parameterized unit tests using `pytest` to verify the correctness of the LZ78 implementation. The tests cover edge cases like empty strings, single characters, repeated patterns, and complex repetitions.

### CODE REVIEW

File: slp-eval-py/src/slp_eval/lz78.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+from dataclasses import dataclass
+from typing import List, Tuple, Dict, TypeVar, Generic
+
+A = TypeVar('A')
+
+@dataclass
+class LZ78(Generic[A]):
+    """LZ78 Compressor and Decompressor."""
```
Comment (AliaumeL) [inline slp-eval-py/src/slp_eval/lz78.py]: Can you put a reference in a comment to explain from where you got the implementation?

**Review by AliaumeL — COMMENTED** [review]
I believe that it would be better to have a class that *represents* a Lempel-Ziv compressd string rather than a class that contains two methods taking `self` as an argument and never using it. Note that if you are not using self, maybe you want to use decorators such as `@classmethod` or `@staticmethod`.

I would suggest that you write a class `LZ78String` that represents a compressed string.

```python
@dataclass
class LZ78String(Generic[A]):
  """ A compressed string using Lempel-Ziv """
  content : list[Tuple[int, A]]

  @staticmethod
  def from_list(s : list[A]) -> 'LZ78String[A]':
      pass

  def to_list(self) -> list[A]:
      pass
```

This has two advantages. First, you have better types because you can actually talk about compressed strings and not `list[tuple[A,int]]`. Second, the way to interact with your code is clear and discoverable: you export one type, and autocompletion will automatically guide users to `from_list` and `to_list` functions.

@vaibhavsiingh

File: slp-eval-py/src/slp_eval/lz78.py (hunk 1 above)
Comment (vaibhavsiingh) [reply slp-eval-py/src/slp_eval/lz78.py]: I've pushed the requested changes:
Let me know if anything else needs adjusting!
